inject/src/main/java/io/micronaut/inject/writer/BeanDefinitionWriter.java

```diff
@@ -1127,6 +1163,150 @@ public void accept(Map<String, ClassElement> stringClassElementMap) {
+                pushCastToType(adapter, String.class);
+
+                // storing to new local variable
+                int propertyValueIndex = adapter.newLocal(stringType);
+                adapter.storeLocal(propertyValueIndex);
+
+                adapter.loadLocal(propertyValueIndex);
+                Label exceptionLabel = new Label();
+                Label furtherCheckLabel = new Label();
+
+                if (visitData.requiredValue != null || visitData.notEqualsValue != null) {
+                    adapter.ifNull(visitData.requiredValue != null ? exceptionLabel : furtherCheckLabel);
+                    adapter.loadLocal(propertyValueIndex);
+                    adapter.push(visitData.requiredValue != null ? visitData.requiredValue : visitData.notEqualsValue);
+                    // comparing through String.equals
+                    adapter.invokeVirtual(stringType, stringEqualsMethod);
+                    adapter.push(false);
+                    adapter.ifCmp(Type.BOOLEAN_TYPE, visitData.requiredValue != null ?
+                                                         GeneratorAdapter.NE : GeneratorAdapter.EQ, furtherCheckLabel);
+                } else {
+                    adapter.ifNull(exceptionLabel);
+                    adapter.goTo(furtherCheckLabel);
+                }
+
+                adapter.visitLabel(exceptionLabel);
+                // Building DisabledBeanException
+                adapter.newInstance(disabledBeanExceptionType);
+                adapter.dup();
+                // Building string concatenation with exception message
+                adapter.newInstance(stringBuilderType);
+                adapter.dup();
+                adapter.invokeConstructor(stringBuilderType, stringBuilderConstructor);
+                adapter.push("Bean [" + beanType.getClassName() + "] is disabled since property getter [" + propertyGetter.getSimpleName() + "] on bean [" + injectedType.getClassName() + "] returned value ");
+                adapter.invokeVirtual(stringBuilderType, stringBuilderAppendMethod);
+                // loading property value which caused exception
+                adapter.loadLocal(propertyValueIndex);
+                // appending property value to exception message
+                adapter.invokeVirtual(stringBuilderType, stringBuilderAppendMethod);
+                adapter.invokeVirtual(stringBuilderType, toStringMethod);
+                // invoking exception constructor
+                adapter.invokeConstructor(disabledBeanExceptionType, disabledBeanExceptionConstructor);
+                adapter.throwException();
+
+                // label visited when no exception is thrown
+                adapter.visitLabel(furtherCheckLabel);
+                // if check is over returning from method
+                if (isLastProperty) {
+                    adapter.returnValue();
+                }
+            }
+        }
+    }
+
```

Comment:
I admire that you managed to generate working byte code for this logic, but it is quite a bit of logic. I think it makes more sense to move the logic into a method that you call with the data and implement the logic in Java, it will make it more maintainable. We try in general to keep us much logic out of the byte code generation as possible.

Reply:
Simplified this part. Property `value` and `notEquals` comparision is now implemented in separate java method in `AbstractInitializableBeanDefinition`